server/services/TelegramBot/telegramBot.js

```diff
@@ -0,0 +1,109 @@
+const { Telegraf } = require("telegraf");
+const API = require("./API_Bot");
+
+const token = process.env.BOT_TOKEN || "";
+if (token === undefined) {
+  throw new Error("BOT_TOKEN must be provided!");
+}
+
+const PORT = process.env.PORT_BOT || 3002;
+const URL = process.env.URL || "https://spg-testing.herokuapp.com/"; //"https://spg-deploy-feat-telegra-lmixoy.herokuapp.com/";
+
+const bot = new Telegraf(token);
+bot.telegram.setWebhook(`${URL}/bot${token}`);
+bot.startWebhook(`/bot${token}`, null, PORT);
+
+bot.launch();
+
+exports.WriteList = async function () {
+  let users = [];
+  users = await API.getTelegramUsers(URL);
+  console.log("Utenti: " + users);
+  users.map((id) => {
+    writeProductList(id.chatID);
+  });
+};
+
+const getEmoji = (category) => {
+  switch (category) {
+    case "fruit":
+      return "🍓";
+    case "vegetables":
+      return "🥦";
+    case "spreadable creams":
+      return "🥫";
+    case "meat":
+      return "🍖";
+    case "eggs":
+      return "🥚";
+    case "milk":
+      return "🥛";
+
+    default:
+      return "🛒";
+  }
+};
+
+const getLeftQuantity = (lq) => {
+  if (!lq) {
+    return "<s>0  pieces left</s>";
+  }
+  return lq + " pieces left";
+};
+
+//Client sends start and will receive a welcome message,
+bot.command("start", (ctx) => {
+  console.log(ctx.from);
+  API.addTelegramUsers(URL, ctx.chat.id);
```

Comment:
you can consider to insert an try catch and await for this call if you want to be sure to reply only if you save the chatID correctly